pkg/kubelet/cm/bootstrap/cgroupfs/bootstrap.go

```diff
@@ -0,0 +1,352 @@
+func ensureDockerInContainer(cadvisor cadvisor.Interface, oomScoreAdj int, manager *fs.Manager) error {
+	// What container is Docker in?
+	out, err := exec.Command("pidof", "docker").Output()
+	if err != nil {
+		return fmt.Errorf("failed to find pid of Docker container: %v", err)
+	}
+
+	// The output of pidof is a list of pids.
+	// Docker may be forking and thus there would be more than one result.
+	pids := []int{}
+	for _, pidStr := range strings.Split(strings.TrimSpace(string(out)), " ") {
+		pid, err := strconv.Atoi(pidStr)
+		if err != nil {
+			continue
+		}
+		pids = append(pids, pid)
+	}
+
+	// Move if the pid is not already in the desired container.
+	errs := []error{}
+	for _, pid := range pids {
+		cont, err := getContainer(pid)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to find container of PID %d: %v", pid, err))
+		}
+
+		if cont != manager.Cgroups.Name {
+			err = manager.Apply(pid)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("failed to move PID %d (in %q) to %q", pid, cont, manager.Cgroups.Name))
+			}
+		}
+
+		// Also apply oom-score-adj to processes
+		oomAdjuster := oom.NewOOMAdjuster()
+		if err := oomAdjuster.ApplyOOMScoreAdj(pid, oomScoreAdj); err != nil {
+			errs = append(errs, fmt.Errorf("failed to apply oom score %d to PID %d", oomScoreAdj, pid))
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+// Gets the (CPU) container the specified pid is in.
+func getContainer(pid int) (string, error) {
+	cgs, err := cgroups.ParseCgroupFile(fmt.Sprintf("/proc/%d/cgroup", pid))
+	if err != nil {
+		return "", err
+	}
+
+	cg, ok := cgs["cpu"]
+	if ok {
+		return cg, nil
+	}
+
+	return "", cgroups.NewNotFoundError("cpu")
+}
+
+// Ensures the system container is created and all non-kernel threads and process 1
+// without a container are moved to it.
+//
+// The reason of leaving kernel threads at root cgroup is that we don't want to tie the
+// execution of these threads with to-be defined /system quota and create priority inversions.
+//
+// The reason of leaving process 1 at root cgroup is that libcontainer hardcoded on
+// the base cgroup path based on process 1. Please see:
+// https://github.com/kubernetes/kubernetes/issues/12789#issuecomment-132384126
+// for detail explanation.
+func ensureSystemContainer(rootContainer *fs.Manager, manager *fs.Manager) error {
+	// Move non-kernel PIDs to the system container.
+	attemptsRemaining := 10
+	var errs []error
+	for attemptsRemaining >= 0 {
+		// Only keep errors on latest attempt.
+		errs = []error{}
+		attemptsRemaining--
+
+		allPids, err := rootContainer.GetPids()
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to list PIDs for root: %v", err))
+			continue
+		}
+
+		// Remove kernel pids and other protected PIDs (pid 1, PIDs already in system & kubelet containers)
+		pids := make([]int, 0, len(allPids))
+		for _, pid := range allPids {
+			if isKernelPid(pid) {
```

Comment:
cool, just saw this merged that had that change, will update: https://github.com/kubernetes/kubernetes/pull/20395